lec16-data-viz.Rmd

```diff
@@ -1,4 +1,97 @@
+we think should work on it and then review the workflow for addressing the issue.
+
+*Create new issue about adding a file to the sandbox repo*
+
+1. Create your own fork of the online git repository via the GitHub interface.
+2. Click the green `Clone or download` button to copy the git URL.
+3. In RStudio, open a new git project.
+    - `File -> Version Control -> Git`. Paste the URL you copied previously and select where to save the directory.
+4. Click `Create Project`
+    - You can now see the new directory if you navigate to it in your file
+      navigator.
+5. Open a terminal and navigate to the new directory.
+    - You could do the navigation in your file browser and then right click to
+      open a terminal
+        - Right click should be enabled by default with Git.. on Windows. For
+          Mac's Finder, follow [these
+          instructions](https://stackoverflow.com/a/7054045/2166823) (in brief
+          `System Preferences > Keyboard > Shortcuts > Services`).
+6. In RStudio, click `History` under the small Git button menu.
+    - This is similar to typing `git log --oneline --graph` in terminal.
+7. Before we add something to this repository, it is good practice to create a
+   new branch instead of working on the default master branch. Branches allows
+   you to work on features separately and can make it clearer what is what when
+   merging your code with others.
+    - This is the only step that *needs* to be done from the command line.
+      Navigate to the directory just created via RStudio and type `git branch
+      <branchname>`. You can now switch to the new branch in the RStudio
+      interface (you could also type `git checkout -b <branchname>` to create
+      and switch to a branch in one step in terminal).
+8. Create a new text file in RStudio and add some text.
+9. Save the file locally in the git directory that you created via RStudio.
+    - These changes are not saved into your git history until you explicitly
+      tell git to save them.
+10. Before saving (committing) into the git repository, click `Diff <filename>`
+   in the RStudio git menu, to see the differences between your current working
+   directory and what you have saved in the git repository since previously.
+    - This would be the same as typing `git diff <filename>` in the terminal.
+11. The differences are what we expect, so let's click `Stage` (either the
+   checkbox or the button).
+    - This would be the same as typing `git add <filename>` in terminal (adding
+      the file to the "staging area".
+12. Now we can commit our changes, which means saving them into our local git
+   repository. Add a commit message and click `Commit`.
+    - This would be the same as typing `git commit <filename> -m 'Your commit
+      message here'` in terminal.
+    - The reason you are staging first and then committing, instead of just
+      "saving" in one step, is that this gives workflow is more powerful and
```

Comment:
`the workflow more power and more flexibility`

Reply:
more power and flexibility.